ZuluContent/Items/Shields/BaseShield.cs

```diff
@@ -30,71 +34,40 @@ public override void Deserialize(IGenericReader reader)
             int version = reader.ReadInt();
         }
 
-        public override int OnHit(BaseWeapon weapon, int damage)
+        public override double OnHit(BaseWeapon weapon, double damage)
         {
             if (!(Parent is Mobile owner))
                 return damage;
+            
+            var armorRating = ArmorRating;
 
-            double ar = ArmorRating;
-            double chance = (owner.Skills[SkillName.Parry].Value - ar * 2.0) / 100.0;
-
-            if (chance < 0.01)
-                chance = 0.01;
-            /*
-            FORMULA: Displayed AR = ((Parrying Skill * Base AR of Shield) � 200) + 1
-
-            FORMULA: % Chance of Blocking = parry skill - (shieldAR * 2)
-
-            FORMULA: Melee Damage Absorbed = (AR of Shield) / 2 | Archery Damage Absorbed = AR of Shield
-            */
-            if (owner.CheckSkill(SkillName.Parry, chance))
+            if (armorRating > 0)
             {
-                if (weapon.Skill == SkillName.Archery)
-                    damage -= (int) ar;
-                else
-                    damage -= (int) (ar / 2.0);
-
-                if (damage < 0)
-                    damage = 0;
-
-                owner.FixedEffect(0x37B9, 10, 16);
+                var parrySkillVal = owner.Skills[SkillName.Parry].Value;
+                if (owner.ClassContainsSkill(SkillName.Swords, SkillName.Macing, SkillName.Anatomy))
```

Comment:
You shoudln't need this test for a warrior class here, you can just use:
```csharp
var parrySkillVal = owner.Skills.Parry.Value * owner.GetClassModifier(SkillName.Parry);
```

The modifier will be `1.0` if not part of the class.

Reply:
Done